### DISCUSSION TITLE
Improving model for Glaucidium passerinum (Eurasian Pygmy-Owl)

### DISCUSSION THREAD

**Discussion by z8manu8z** [birdnet-team/BirdNET-Analyzer#334] [Q&A]
I'm trying to get Birdnet detect small notes of the Eurasian Pygmy-Owl (not detected by current software).

I extracted around ten sequences of 1 to 3 seconds containing precisely the notes to search for, also extracted around ten files containing the background noise and launched Birdnet Training (I use Windows GUI version).

![2024-05-14 093629](https://github.com/kahst/BirdNET-Analyzer/assets/24700386/9646b265-2efb-4e4b-b00b-1ee5ca7768dd)   ![2024-05-14 093702](https://github.com/kahst/BirdNET-Analyzer/assets/24700386/fc189b17-2ebb-4550-93ce-64064f21598a)   ![2024-05-14 093811](https://github.com/kahst/BirdNET-Analyzer/assets/24700386/195fb5f5-9326-4fd7-97fb-937e555e742b)

Despite all my attempts, impossible to detect the patterns in the attached images. I often have false positives or no detection.
I tried leaving the extracts unchanged (image 1 and 2) but also removing unwanted frequencies with Audacity (image 3).

I even tried to lower Birdnet's sensitivity sliders to make it detect these patterns even tagged with another species, in vain.
Do you have any idea how to train the model to detect these notes ?

Thanks a lot for your help ;-)

**Comment by DD4WH** [comment]
Very nice to hear you are also interested in Pygmy Owl!
We have been using BirdNET to detect the species for a few years now with very good results. It detects the male and female hooting and also the scales, but for the latter it is not so sensitive. Also, the female high frequency sree calls are detected, but with a lot of false positives.  
I am not sure what the calls are that you posted, because the spectrograms lack a frequency scale. Could you have a look here in this article and tell us which calls they correspond to? 

https://xeno-canto.org/article/277

I think your approach has two potential flaws:

1. Your training calls have many other simultaneously singing birds, so you train the classifier to identify a mixture of birds
2. 10 or so recordings are not enough, try to train with at least one hundred recordings 

Would be nice to know more about your approach!

**Reply by z8manu8z** [reply]
Hello @DD4WH  and thanks a lot for your reply !
Here is a capture with a frequency scale, sorry for this oversight ;-)

![2024-05-15 084944](https://github.com/kahst/BirdNET-Analyzer/assets/24700386/c00424a4-40f1-4453-8913-743638429923)

I read the very interesting article on xeno-canto.
The notes at 1600 Hz are those of the male (very short contact calls), and those at 5500 + pending HF 11000 Hz are the calls of the female (female soliciting call https://xeno-canto.org/789299)

I know my approach is not perfect but I don't have a lot of files with notes of this type, where the animals are very discreet.
To manage the case of simultaneous birdsong, do I have to "clean" the files before launching training phase by removing unwanted frequencies ?

I attach a fragment of the file + audacity labels on which I base myself to teach the model to detect these short songs, can you tell me if you manage to detect something?

[ac_2024-04-08_07-20-38.zip](https://github.com/kahst/BirdNET-Analyzer/files/15317723/ac_2024-04-08_07-20-38.zip)

Thanks in advance :-)

**Reply by DD4WH** [reply]
Hi!
Thanks for providing the original recording! 

* Did you really see the bird and can you confirm that those calls are really male and female Pygmy Owl ?
* for my ears, neither of the calls sound like Pygmy Owl. I maybe wrong, but if your calls are really Glaucidium passerinum, the calls you recorded are very unusual and do not correspond to the calls I know
* however, even if your recording contains Pygmy Owl, you cannot train a CNN classifier with only a few recordings
* if you want to train a classifier with very few or only one "template", you could try the package "monitoR" in the statistic software R -> I have achieved good results for simple bird calls with that technique. However, it is not suitable for more complex sounds

https://www.tandfonline.com/doi/full/10.1080/09524622.2016.1138415

Good luck!

**Reply by z8manu8z** [reply]
Hello @DD4WH !
I can confirm that they are indeed Pygmy Owls ;-) A few seconds later we hear the calls intensify, followed by a copulation. There is absolutely no possible doubt.
I'm in France, maybe the songs aren't quite the same! What region are you in?

Thanks for the "monitoR" tip, I'll try ;-)
By the way, do you have a tutorial or quick tips on hand for carrying out this type of research with monitoR?

My current problem is that for the past weeks I have lost track of the couple that I followed over the past months, and I am trying to find them using microphones. If I don't have software to help me, I have to analyze dozens of hours of recording ...

Thanks again for your help.
